src/components/generic/Alert.tsx

```diff
@@ -1,81 +1,154 @@
+    case 'success':
+      return <CheckCircle {...iconProps} />;
     case 'error':
-      return <WarningCircle />
-    case 'warning':
-      return <Warning />
+      return <XCircle {...iconProps} />;
     case 'info':
-      return <Info />
-    case 'success':
-      return <CheckCircle />
+      return <Info {...iconProps} />;
+    case 'warning':
+      return <Warning {...iconProps} />;
+    default:
+      return <span style={{ fontSize: '5em' }}>❔</span>;
   }
-}
+};
 
-const Alert = ({
-  severity,
-  title,
-  description,
-  marginTop,
-  marginButton,
-  className,
-  isScrollIntoView,
-}: AlertProps) => {
-  const alertRef = useRef<HTMLDivElement>(null)
+const Alert = ({ severity, title, description, marginTop, className, style, onDismiss }: AlertProps) => (
+  <div
+    className={cx('alert-card', `alert-${severity}`, className, marginTop && 'marginTop')}
+    style={style}
+    onClick={() => onDismiss?.()}
+  >
+    <div className={cx('alert-icon')}>{getAlertIcon(severity)}</div>
+    <div className={cx('alert-content-container')}>
+      {title && <div className={cx('alert-title')}>{title}</div>}
+      {description && <div className={cx('alert-content')}>{description}</div>}
+    </div>
+  </div>
+);
 
-  useEffect(() => {
-    if (isScrollIntoView && alertRef.current) {
-      alertRef.current.scrollIntoView({ behavior: 'smooth' })
+export const AlertsPromptProvider = ({ children }: { children: ReactNode }) => {
+  const [alerts, setAlerts] = useState<AlertType[]>([]);
+  const [fontSize, setFontSize] = useState('1rem');
+  const alertTimeouts = useRef<{ [key: string]: NodeJS.Timeout }>({});
+
+  const adjustFontSize = () => {
+    if (alerts.length === 0) {
+      setFontSize('1rem');
+      return;
+    }
+
+    const longestMessageLength = Math.max(
+      ...alerts.map(
+        (alert) => `${alert.title ?? ''} ${alert.description ?? ''}`.length
+      )
+    );
+
+    let calculatedFontSize = '1rem';
+    if (longestMessageLength > 100) {
+      calculatedFontSize = '0.8rem'; 
+    } else if (longestMessageLength > 50) {
+      calculatedFontSize = '0.875rem'; 
+    }
+
+    setFontSize(calculatedFontSize);
+  };
+
+  const showAlert = ({ title, severity, description }: Omit<AlertType, 'id'>) => {
+    setAlerts((prev) => {
+      if (prev.some((alert) => alert.severity === severity)) {
+        return prev;
+      }
+      const id = `${Date.now()}-${Math.random()}`;
+      const newAlert = { id, title, severity, description };
+
+      alertTimeouts.current[id] = setTimeout(() => {
+        removeAlert(id);
+      }, 10000);
+
+      return [...prev, newAlert];
+    });
+  };
+
+  const removeAlert = (id: string) => {
+    setAlerts((prev) => prev.filter((alert) => alert.id !== id));
+    if (alertTimeouts.current[id]) {
+      clearTimeout(alertTimeouts.current[id]);
+      delete alertTimeouts.current[id];
     }
-  }, [isScrollIntoView])
+  };
+
+  useEffect(() => {
+    adjustFontSize();
+  }, [alerts]);
```

Comment:
Since this effect is calling just one function, I suggest the following changes:
1. Move the effect to be directly below the function.
2. Use the function reference rather than wrapping it in another arrow function: `useEffect(adjustFontSize, [alerts])`